lib/tests/formulae_dependents.rb

```diff
@@ -171,14 +171,21 @@ def install_dependent(dependent, testable_dependents, args:, build_from_source:
         if testable_dependents.include? dependent
           test "brew", "install", "--only-dependencies", "--include-test", dependent.full_name
 
-          dependent.recursive_dependencies.each do |dependency|
-            next if dependency.build? && !dependency.test?
+          # Traverse the dependency tree to check for formulae we need to link
+          dependencies_to_link = Dependency.expand(
+            dependent,
+            cache_key: "test-bot-link",
```

Comment:
```suggestion
            cache_key: "test-bot-link-#{dependent.full_name}",
```

This is still `dependent`-specific because of `dep_dependent == dependent`